src/cmp.rs

```diff
@@ -0,0 +1,267 @@
+    pub removed: Vec<String>,
+}
+
+///
+/// Compares 2 icon fonts `lhs`, `rhs` and returns names for
+/// `CompareResult::added`: icons in `lhs` but not in `rhs`.
+/// `CompareResult::modified`: icons in both `lhs` and `rhs` that draws differently.
+/// `CompareResult::removed`: icons not in `lhs` but in `rhs`.
+///
+pub fn cmp(lhs: &FontRef, rhs: &FontRef) -> Result<CompareResult, IconResolutionError> {
+    let lhs_icons = get_icons(lhs)?;
+    let rhs_icons = get_icons(rhs)?;
+    let lhs_icons: HashMap<String, GlyphId> = map_by_names(lhs_icons);
+    let rhs_icons: HashMap<String, GlyphId> = map_by_names(rhs_icons);
+    let added = in_first_but_not_second(&lhs_icons, &rhs_icons);
+    let removed = in_first_but_not_second(&rhs_icons, &lhs_icons);
+    let modified = diff_glyphs(lhs_icons, rhs_icons, lhs, rhs)?;
+    Ok(CompareResult {
+        added,
+        modified,
+        removed,
+    })
+}
+
+fn diff_glyphs(
+    lhs_icons: HashMap<String, GlyphId>,
+    rhs_icons: HashMap<String, GlyphId>,
+    lhs: &FontRef,
+    rhs: &FontRef,
+) -> Result<Vec<String>, IconResolutionError> {
+    let lhs_outlines = Tables::new(lhs)?;
+    let rhs_outlines = Tables::new(rhs)?;
+    let common: Vec<(String, GlyphId, GlyphId)> = lhs_icons
+        .into_iter()
+        .filter_map(|(k, v)| rhs_icons.get(&k).map(|r_gid| (k, v, *r_gid)))
+        .collect();
+    let result = common
+        .par_iter()
+        .map(|(name, lhs_gid, rhs_gid)| {
+            let mut lhs_closure: Vec<_> = lhs
+                .gsub()?
+                .closure_glyphs([*lhs_gid].into())?
+                .into_iter()
+                .collect();
+            let mut rhs_closure: Vec<_> = rhs
+                .gsub()?
+                .closure_glyphs([*rhs_gid].into())?
+                .into_iter()
+                .collect();
+            if lhs_closure.len() != rhs_closure.len() {
+                return Ok::<String, IconResolutionError>(name.to_string());
+            }
+            lhs_closure.sort();
+            rhs_closure.sort();
+            for (lhs_gid, rhs_gid) in lhs_closure.iter().zip(rhs_closure.iter()) {
+                if !eq(&lhs_outlines, &rhs_outlines, *lhs_gid, *rhs_gid)? {
+                    return Ok(name.to_string());
+                }
+            }
+            Ok(String::from(""))
+        })
+        .collect::<Result<Vec<_>, _>>()?
+        .into_iter()
+        .filter(|f| !f.is_empty());
+
+    Ok(result.collect())
+}
+
+struct Tables<'a> {
+    gvar: Option<Gvar<'a>>,
+    outlines: OutlineGlyphCollection<'a>,
+}
+
+impl<'a> Tables<'a> {
+    fn new(font: &'a FontRef) -> Result<Tables<'a>, ReadError> {
+        Ok(Tables {
+            gvar: font.gvar().ok(),
+            outlines: font.outline_glyphs(),
+        })
+    }
+}
+
+fn eq(
+    lhs: &Tables,
+    rhs: &Tables,
+    lhs_gid: GlyphId,
+    rhs_gid: GlyphId,
+) -> Result<bool, IconResolutionError> {
+    if lhs.gvar.is_some() ^ rhs.gvar.is_some() {
```

Comment:
I kinda like the xor here but a simple `!=` should suffice and will be less likely to cause confusion.